src/components/TrainingComponents/TrainingVideo.tsx

```diff
@@ -1,46 +1,88 @@
-import * as React from "react";
-import { DefaultPlayer as Video } from "react-html5video/dist";
-import "react-html5video/dist/styles.css";
-import * as PropTypes from "prop-types";
+import * as React from 'react'
+import { DefaultPlayer as Video } from 'react-html5video/dist'
+import 'react-html5video/dist/styles.css'
+import * as PropTypes from 'prop-types'
+import { addWatchedVideos } from '../../state/actions/watched-videos'
+import { connect } from 'react-redux'
 
 interface Props {
-  videoUrl: string
+    videoUrl: string
+    watchedVideos: Array<string>
+}
+
+interface State {
+    watchedVideos: Array
 }
 
 /**
  * specifies controls and default settings for demo video on landing page
  * @class DemoVideo
  */
-class TrainingVideo extends React.Component<Props, {}> {
-  static propTypes = {
-    videoUrl: PropTypes.string.isRequired
-  }
-  /**
-   * render function
-   * @return {ReactNode}
-   */
-  render(): React.ReactNode {
-    return (
-      <Video
-        loop
-        muted
-        controls={["PlayPause", "Seek", "Time", "Volume", "Fullscreen"]}
-        poster="http://sourceposter.jpg"
-        onCanPlayThrough={(): void => {
-          // Do stuff
-        }}
-      >
-        <source src={this.props.videoUrl} type="video/mp4" />
-        <track
-          label="English"
-          kind="subtitles"
-          srcLang="en"
-          src="http://source.vtt"
-          default
-        />
-      </Video>
-    );
-  }
+class TrainingVideo extends React.Component<Props, State> {
+    /**
+     * @param {Props} props
+     */
+    constructor(props: Props) {
+        // section -> one of ('transition','climate','ac',etc...)
+        super(props)
+        this.state = {
+            videoUrl: [],
+            autoPlay: false,
+        }
+    }
+
+    static propTypes = {
+        videoUrl: PropTypes.string.isRequired,
+    }
```

Comment:
With power of TypeScript using `prop-types` is redundant - `Props` interface already has the type info and types are thus checked in the compile time rather than in runtime. Let's remove that block
```suggestion
```